### PR TITLE
User show page on React

### PR DESCRIPTION
**Pull request by Axedez**
[Trello Task](https://trello.com/c/9YU0VagQ/92-show-user-page-on-react)

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
- [x] screenshot of feature is added if task has UI
ezgif com-gif-maker (1)

### CODE REVIEW

**Review by Ukhanskyi — APPROVED** [review]

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Api::V1::UsersController < Api::ApiController
+  def index
+    @users = User.all
```
Comment (pereslop) [inline app/controllers/api/v1/users_controller.rb:3]: it is bad to fetch all users, what if you will have 10000000 users, you will load all of them?

Comment (Axedez) [reply app/controllers/api/v1/users_controller.rb:3]: I need list of all users for partial to show them on index page

**Review by IvanDanyliv — APPROVED** [review]

File: app/javascript/components/users/show/ShowUser.jsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        address: []
+      }
+    };
+    this.osbbInfo = this.osbbInfo.bind(this)
+    this.addressInfo = this.addressInfo.bind(this)
+  };
+
+  componentDidMount(){
+    const id = window.location.pathname.split('/')
+    fetch('/api/v1/users/' + id[id.length - 1] + '.json')
```
Comment (chu4) [inline app/javascript/components/users/show/ShowUser.jsx:23]: why dont you send id in props?
looks very weird here

File: app/views/api/v1/users/_user.json.jbuilder [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+json.role user.role.capitalize
+
+json.address do
+  json.country user.address.country
+  json.state user.address.state
+  json.city user.address.city
+  json.street user.address.street
+end
+
+if user.avatar?
```
Comment (chu4) [inline app/views/api/v1/users/_user.json.jbuilder:11]: probably should be a user model method to return existing avatar or default

File: app/javascript/components/users/show/Control.jsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        this.goBack()
+      })
+      .catch((error) => {
+        alert(error);
+      });
+    };
+
+  render() {
+    const svgIconReturn = (
+      <svg width="1em"
```
Comment (vasya539) [inline app/javascript/components/users/show/Control.jsx:30]: можеш svg винести в файл і імпортнути

File: app/javascript/components/users/show/Control.jsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+           fill="currentColor"
+           xmlns="http://www.w3.org/2000/svg"
+      >
+        <path fillRule="evenodd"
+              d="M14.5 1.5a.5.5 0 0 1 .5.5v4.8a2.5 2.5 0 0 1-2.5 2.5H2.707l3.347 3.346a.5.5 0 0 1-.708.708l-4.2-4.2a.5.5 0 0 1 0-.708l4-4a.5.5 0 1 1 .708.708L2.707 8.3H12.5A1.5 1.5 0 0 0 14 6.8V2a.5.5 0 0 1 .5-.5z"
+        />
+      </svg>
+    );
+
+    const svgIconDelete = (
```
Comment (vasya539) [inline app/javascript/components/users/show/Control.jsx:43]: можна повиносити в якийсь файл де будуть лежати подібні речі і просто їх імпортити, це напевно буде переюзуватись багато де

File: app/javascript/components/users/show/ShowUser.jsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+            className='title'>User info:
+        </h1>
+        <div className='show-box'>
+          <div data-aos="fade-up"
+               data-aos-delay="1800"
+               data-aos-offset="0"
+               data-aos-duration="1500"
+               className='user-info'
+          >
+            <InfoHelper name='Username:' value={ this.state.user.full_name } />
```
Comment (vasya539) [inline app/javascript/components/users/show/ShowUser.jsx:72]: куча однакових інфохелперів які просто міняють значення, можна це через map

**Review by chu4 — APPROVED** [review]

**Pull request merged**